frontend/src/metabase/data-grid/components/DataGrid/DataGrid.unit.spec.tsx

```diff
@@ -0,0 +1,338 @@
+
+interface TestDataGridProps {
+  onHeaderCellClick?: (
+    event: MouseEvent<HTMLDivElement>,
+    columnId?: string,
+  ) => void;
+  onBodyCellClick?: (
+    event: MouseEvent<HTMLDivElement>,
+    rowIndex: number,
+    columnId: string,
+  ) => void;
+  onAddColumnClick?: () => void;
+  onColumnResize?: (columnSizing: ColumnSizingState) => void;
+  onColumnReorder?: (columnIds: string[]) => void;
+  initialColumnSizing?: Record<string, number>;
+  initialColumnOrder?: string[];
+  enableRowId?: boolean;
+  sortableColumns?: boolean;
+  wrapableColumns?: string[];
+}
+
+const TestDataGrid = ({
+  onHeaderCellClick,
+  onBodyCellClick,
+  onAddColumnClick,
+  onColumnResize,
+  onColumnReorder,
+  initialColumnSizing = DEFAULT_COLUMN_SIZING,
+  initialColumnOrder = DEFAULT_COLUMN_ORDER,
+  enableRowId = false,
+  sortableColumns = false,
+  wrapableColumns = [],
+}: TestDataGridProps) => {
+  const columns: ColumnOptions<SampleDataType>[] = useMemo(
+    () => [
+      {
+        id: "id",
+        name: "ID",
+        accessorFn: row => row.id,
+        align: "right",
+        cellVariant: "pill",
+        sortDirection: sortableColumns ? "desc" : undefined,
+      },
+      {
+        id: "name",
+        name: "Name",
+        accessorFn: row => row.name,
+        sortDirection: sortableColumns ? "asc" : undefined,
+        wrap: wrapableColumns.includes("name"),
+      },
+      {
+        id: "category",
+        name: "Category",
+        accessorFn: row => row.category,
+        wrap: wrapableColumns.includes("category"),
+        getBackgroundColor: value => {
+          switch (value) {
+            case "Electronics":
+              return "rgb(230, 247, 255)";
+            case "Clothing":
+              return "rgb(246, 255, 237)";
+            case "Books":
+              return "rgb(255, 247, 230)";
+            case "Food":
+              return "rgb(255, 241, 240)";
+            default:
+              return "";
+          }
+        },
+      },
+      {
+        id: "price",
+        name: "Price",
+        accessorFn: row => row.price,
+        formatter: value => `$${value}`,
+        align: "right",
+        wrap: wrapableColumns.includes("price"),
+      },
+    ],
+    [sortableColumns, wrapableColumns],
+  );
+
+  const rowId: RowIdColumnOptions | undefined = useMemo(
+    () =>
+      enableRowId
+        ? {
+            variant: "indexExpand",
+          }
+        : undefined,
+    [enableRowId],
+  );
+
+  const tableProps = useDataGridInstance({
+    data: sampleData,
+    columnsOptions: columns,
+    columnOrder: initialColumnOrder,
+    columnSizing: initialColumnSizing,
+    onColumnReorder,
+    onColumnResize,
+    rowId,
+  });
+
+  return (
+    <DataGrid
+      {...tableProps}
+      onHeaderCellClick={onHeaderCellClick}
+      onBodyCellClick={onBodyCellClick}
+      onAddColumnClick={onAddColumnClick}
+    />
+  );
+};
+
+describe("DataGrid", () => {
```

Comment:
Probably worth adding a test case for the DnD functionality

Reply:
This turned to be quite challenging in jsdom and I think it won't be a meaningful test. If we don't already have an E2E spec for dnd in Table viz which will use this grid I will add it there